File: TaskManager/ClientApp/app/components/shared/models/taskfile.ts
export class TaskFile {
	taskFileID: number;
	taskID: number;
	fileName: string;
	isAdded: boolean;
	isDeleted: boolean;
	fileData: any;

	constructor() {
		this.taskFileID = 0;
		this.taskID = 0;
		this.fileName = "";
		this.isAdded = false;
		this.isDeleted = false;
	}
}

export interface IOneTaskFile {
	taskFile: TaskFile;
}

File: TaskManager/Models/TaskFile.cs
using System.Collections.Generic;

namespace TaskManager.Models
{
    public class TaskFile
    {
		public int TaskFileID { get; set; }
		public int TaskID { get; set; }
		public string FileName { get; set; }
	}
	public class TaskFileSend : TaskFile
	{
		public bool IsAdded { get; set; }
		public bool IsDeleted { get; set; }
	}

	public class TaskFilesList
	{
		public List<TaskFileSend> taskFiles { get; set; }
	}
}
